controllers/auth.js

```diff
@@ -4,98 +4,70 @@ const jwt = require("jsonwebtoken");
 const { JWT_SECRET } = require("../config/config");
 const { User } = require("../models");
 
-
-exports.userSignUp = (req, res) => {
-
+exports.userSignUp = async (req, res) => {
     const { name, password, email, resumptionDate, phoneNumber, homeAddress } = req.body;
-    User
-        .find({ email })
-        .exec()
-        .then( result => {
-            if (result.length >= 1) {
-                return res.status(409).json({
-                    message: 'User already exists.'
-                })
-            } else {
-                bcrypt.hash(password, 10, (err, hash) => {
-                    if (err) {
-                        return res.status(500).json({
-                            error: 'Error occurred.'
-                        })
-                    } else {
-                        const user = new User({
-                            name, 
-                            password: hash, 
-                            email, 
-                            resumptionDate, 
-                            phoneNumber, 
-                            homeAddress 
-                        })
-                        user
-                            .save()
-                            .then( result => {
-                                console.log('User signed up successfully');
-                                return res.status(201).json({ 
-                                    message: 'User(staff) signed up successfully',
-                                    result
-                                });
-                            })
-                            .catch( err => {
-                                console.log('Server error');
-                                return res.status(500).json({
-                                    message: 'Problem occurred while processing your request..'
-                                })
-                            })
-                    }
-                })
-            }
-        })
-
-}
+    const userExists = await User.findOne({ email });
```

Comment:
This variable "userExists" is confusing, you should use something like user, userExists makes it seem like it holds a boolean value.

Reply:
ok, i'd change it